Automattic/wp-calypso
# Checkout: Move Bancontact payment method to wpcom-checkout

#### Changes proposed in this Pull Request

This PR moves the Bancontact payment method from the `@automattic/composite-checkout` package into the `@automattic/wpcom-checkout` package.

The `composite-checkout` package is intended to be a fairly generic checkout-building toolkit. As part of this we originally envisioned it containing various payment methods, but that quickly became a problem as it became clear that the payment methods we use on WordPress.com are pretty specific to our platform. To reduce leaky abstractions, we moved many of our payment methods over to calypso, but some have remained because the cost of moving them seemed too high for the benefit.

As we prepare to publish version 1.0.0 of the `@automattic/composite-checkout` package on npm, I think it's a good time to clean up extraneous parts of the package. The payment methods are the only part of the package that are not fully TypeScript and are also still pretty specific to our WordPress.com infrastructure. Eventually it would be nice to convert them all to TypeScript and put them in a package like `@automattic/wpcom-checkout` (which is intended to be WordPress.com specific). This PR converts one such payment method and moves it there.

<img width="495" alt="Screen Shot 2021-04-12 at 7 12 20 PM" src="https://user-images.githubusercontent.com/2036909/114473856-0af9c080-9bc3-11eb-92f4-f82d45b589a7.png">


#### Testing instructions

- Apply D45443-code to your sandbox to force Bancontact to be available.
- Set your user currency to EUR.
- Make a new purchase in checkout and select Bancontact as the payment method, entering any string in the "name" field.
- Verify that you are redirected to the Stripe test page, that it reads "Bancontact", and that the total is correct. (No need to complete the payment.)
